client/lib/cart/store/index.js

```diff
@@ -71,8 +72,22 @@ function emitChange() {
 }
 
 function update( changeFunction ) {
+	const reduxState = reduxGetState();
+	const isFetchingProductsList = isProductsListFetching( reduxState );
+	const productsList = getProductsList( reduxState );
+
+	if ( ! productsList ) {
+		if ( ! isFetchingProductsList ) {
+			reduxDispatch( requestProductsList );
+		}
+
+		setTimeout( () => update( changeFunction ), 300 );
```

Comment:
why a timeout? could we hook into the reduxStore's change event instead?

Reply:
The legacy `productsList` usage didn't wait for the list to update. It just triggered the request (by calling `get()`) and worked with the potentially empty object it returned.

@lamosty What was your intent here? To fix a bug?